### PR TITLE
add error messages for failing test results

### PR DESCRIPTION
**Pull request by notshivansh**

### CODE REVIEW

File: apps/testing/src/main/java/com/akto/testing/Main.java [hunk 1]
```diff
@@ -196,6 +197,42 @@ public static void main(String[] args) throws InterruptedException {
                         Filters.eq("_id", testingRun.getId()),  completedUpdate
                 );
 
+                if(summaryId != null){
+                    TestingRunResultSummary testingRunResultSummary = TestingRunResultSummariesDao.instance.findOne(
+                        Filters.eq(TestingRunResultSummary.ID, summaryId));
+                    if(testingRunResultSummary.getState() == TestingRun.State.RUNNING
+                            || testingRunResultSummary.getState() == TestingRun.State.SCHEDULED){
+
```
Comment (ayushaga14) [inline apps/testing/src/main/java/com/akto/testing/Main.java:205]: why are we recalculating all this(COUNT_ISSUES etc), if it's already being done inside apiwiseinit

File: apps/testing/src/main/java/com/akto/testing/Main.java [hunk 2]
```diff
@@ -196,6 +197,42 @@ public static void main(String[] args) throws InterruptedException {
...
+                            totalApis = testingRun.getTestingEndpoints().returnApis().size();
+                        } catch (Exception e) {
+                            totalApis = 0;
+                        }
+
+                        Bson updates = Updates.combine(
+                                Updates.set(TestingRunResultSummary.END_TIMESTAMP, Context.now()),
+                                Updates.set(TestingRunResultSummary.STATE, State.COMPLETED),
+                                Updates.set(TestingRunResultSummary.COUNT_ISSUES, totalCountIssues),
+                                Updates.set(TestingRunResultSummary.TOTAL_APIS, totalApis),
```
Comment (ayushaga14) [inline apps/testing/src/main/java/com/akto/testing/Main.java:227]: what's the use of total_apis

File: apps/testing/src/main/java/com/akto/testing/Main.java [hunk 3]
```diff
@@ -196,6 +197,42 @@ public static void main(String[] args) throws InterruptedException {
                         Filters.eq("_id", testingRun.getId()),  completedUpdate
                 );
 
+                if(summaryId != null){
+                    TestingRunResultSummary testingRunResultSummary = TestingRunResultSummariesDao.instance.findOne(
+                        Filters.eq(TestingRunResultSummary.ID, summaryId));
+                    if(testingRunResultSummary.getState() == TestingRun.State.RUNNING
```
Comment (ayushaga14) [inline apps/testing/src/main/java/com/akto/testing/Main.java:203]: why on scheduled?

File: apps/testing/src/main/java/com/akto/testing/Main.java (hunk 1 above)
Comment (ayushaga14) [reply apps/testing/src/main/java/com/akto/testing/Main.java:205]: let's not do this in this way. Let's identify what are the possible reasons why status is not getting updated and handle in properly. No additions in main.java

**Review by ayushaga14 — CHANGES_REQUESTED** [review]
check comments

File: apps/testing/src/main/java/com/akto/testing/Main.java (hunk 1 above)
Comment (notshivansh) [reply apps/testing/src/main/java/com/akto/testing/Main.java:205]: Sure, let me refractor this PR to only add error messages for failing test runs.

**Review by ayushaga14 — APPROVED** [review]

**Pull request merged**